engine/src/eventqueue.cpp

```diff
@@ -650,6 +636,33 @@ static MCStack *MCEventQueueGetEventStack(MCEvent *p_event)
 	return nil;
 }
 
+static void MCEventQueueDestroyEvent(MCEvent *p_event)
+{
+    if (p_event -> type == kMCEventTypeImeCompose)
+        MCMemoryDeleteArray(p_event -> ime . compose . chars);
+    else if (p_event -> type == kMCEventTypeUpdateMenu)
+    {
+        MCObjectHandle t_handle = p_event->menu.target;
+        t_handle.ExternalRelease();
```

Comment:
So, obviously we need to construct an `MCObjectHandle` from the `target` pointer here so that it can be properly released.  However, it's not actually obvious that an `MCObjectHandle` _needs_ to be constructed here just by inspection.  So what about doing something like:

```C++
MCObjectHandle(p_event->menu.target).ExternalRelease();
```

This would make it clearer that it's actually pretty important to construct the `MCObjectHandle`.

Obviously this code is correct as it is, this is just a suggestion to make it more "obviously correct".